promdump: record the flags provided by the user

ionthegeek opened the pull request:
This commit adds a func that loops through the list of flags parsed by the flag module and records each flag and value in the log.

The yba_api_token flag has special handling to avoid recording the token value.

Commit:
promdump: record the flags provided by the user

This commit adds a func that loops through the list of flags parsed by the flag module and records each flag and value in the log.

The yba_api_token flag has special handling to avoid recording the token value.

cigoldstein reviewed (approved):
LGTM

cigoldstein commented on promdump/promdump.go:
```diff
@@ -883,6 +883,16 @@ func main() {
+			logger.Printf("\t--%s=%v", f.Name, f.Value)
+		}
+	})
+
```
As a nitpicky suggestion, I think it would be preferable to print all of the flags on a single line, instead of on multiple lines. This would make it easier to parse the flags from the logfile using `grep` or something similar.

ionthegeek replied:
Done.

Commit:
promdump: record the flags provided by the user

This commit adds a func that loops through the list of flags parsed by the flag module and records each flag and value in the log.

The flags are joined using a string builder and logged as a single line for ease of greppage.

The yba_api_token flag has special handling to avoid recording the token value. This masking uses a generalized mechanism, so sensitive flags added in future can be trivially masked by adding them to the sensitiveFlags slice by name.

Commit:
promdump: record the flags provided by the user

This commit adds a func that loops through the list of flags parsed by the flag module and records each flag and value in the log.

The flags are joined using a string builder and logged as a single line for ease of greppage. Fixed typo in log message.

The yba_api_token flag has special handling to avoid recording the token value. This masking uses a generalized mechanism, so sensitive flags added in future can be trivially masked by adding them to the sensitiveFlags slice by name.
